### DISCUSSION TITLE
Supply/upload_to_play_store returns a success message but nothing is uploaded on the play store

### DISCUSSION THREAD

**Discussion by alexandrebedel** [fastlane/fastlane#21964] [Q&A] [answered]
Hello everyone,

I'm encountering an issue with supply/upload_to_play_store action. Despite receiving a success message upon uploading my Android app to the Play Store, I'm unable to find it on the store.

<details>

<summary>Here's the lane i'm using</summary>

```ruby
  desc 'Builds the Android APP for release'
  lane :release do |options|
    release = patch_versions(version: options[:version], build_type: 'Release', hotfix: options[:hotfix])

    update_version_file(release)
    commit_files(release) if UI.confirm('Do you want to commit the version files?')
    build_ionic(env: 'production')
    gradle(
      task: 'clean',
      project_dir: './android/',
      print_command_output: false
    )
    gradle(
      task: 'bundle',
      build_type: 'Release',
      project_dir: './android/',
      print_command_output: false
    )

    if UI.confirm("Are you sure to upload the version #{release[:version]} (#{release[:code]}) to the store?")
      upload_to_play_store(
        release_status: 'draft',
        skip_upload_apk: true,
        validate_only: true,
        skip_upload_metadata: true,
        skip_upload_images: true,
        skip_upload_screenshots: true
      )
    end
  end
```
</details>

<details>

<summary>Here's the fastlane log</summary>

```
[14:42:51]: ----------------------------------
[14:42:51]: --- Step: upload_to_play_store ---
[14:42:51]: ----------------------------------

+---------------------------------------------------------------------------------------------------------------------------+
|                                                Summary for supply 2.220.0                                                 |
+------------------------------------+--------------------------------------------------------------------------------------+
| release_status                     | draft                                                                                |
| skip_upload_apk                    | true                                                                                 |
| validate_only                      | true                                                                                 |
| skip_upload_metadata               | true                                                                                 |
| skip_upload_images                 | true                                                                                 |
| skip_upload_screenshots            | true                                                                                 |
| aab                                | /.../android/app/build/outputs/bundle/release/app-release.aab                                                               |
| package_name                       | ....                                                                           |
| track                              | production                                                                           |
| metadata_path                      | ./fastlane/metadata/android                                                          |
| json_key                           | ./android/play-store-key.json                                                        |
| skip_upload_aab                    | false                                                                                |
| skip_upload_changelogs             | false                                                                                |
| sync_image_upload                  | false                                                                                |
| track_promote_release_status       | completed                                                                            |
| check_superseded_tracks            | false                                                                                |
| timeout                            | 300                                                                                  |
| deactivate_on_promote              | true                                                                                 |
| changes_not_sent_for_review        | false                                                                                |
| rescue_changes_not_sent_for_review | true                                                                                 |
| ack_bundle_installation_warning    | false                                                                                |
+------------------------------------+--------------------------------------------------------------------------------------+

[14:42:52]: Preparing aab at path '/.../android/app/build/outputs/bundle/release/app-release.aab' for upload...
[14:43:06]: Updating track 'production'...
[14:43:07]: Preparing uploads for language 'en-US'...
[14:43:07]: Preparing uploads for language 'fr-FR'...
[14:43:07]: Updating changelog for '24080200' and language 'en-US' to default changelog...
[14:43:07]: Uploaded all items for language 'en-US'... (0.266243 secs)
[14:43:07]: Updating changelog for '24080200' and language 'fr-FR' to default changelog...
[14:43:07]: Uploaded all items for language 'fr-FR'... (0.270209 secs)
[14:43:08]: Validating all changes with Google Play...
[14:43:09]: Successfully validated the upload to Google Play

+----------------------------------------------------------------------------+
|                              fastlane summary                              |
+------+-------------------------------------------------------+-------------+
| Step | Action                                                | Time (in s) |
+------+-------------------------------------------------------+-------------+
| 1    | default_platform                                      | 0           |
| 2    | Switch to android patch_versions lane                 | 0           |
| 3    | android_set_version_name                              | 0           |
| 4    | android_set_version_code                              | 0           |
| 5    | Switch to android update_version_file lane            | 0           |
| 6    | git log --tags --simplify-by-decoration               | 0           |
|      | --pretty='format:%ci                                  |             |
| 7    | get_build_number_repository                           | 0           |
| 8    | git show -s --format=%ci                              | 0           |
| 9    | git_branch                                            | 0           |
| 10   | Switch to android build_ionic lane                    | 0           |
| 11   | ionic capacitor build ios --configuration production  | 26          |
|      | --no-ope                                              |             |
| 12   | clean                                                 | 1           |
| 13   | bundleRelease                                         | 9           |
| 14   | upload_to_play_store                                  | 17          |
+------+-------------------------------------------------------+-------------+

[14:43:09]: fastlane.tools finished successfully 🎉 
```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 14.4.1                                      |
| Ruby                        | 3.2.2                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.39.3 (Apple Git-146)          |
| Installation Source         | ~/.rvm/gems/ruby-3.2.2/bin/fastlane         |
| Host                        | macOS 14.4.1 (23E224)                       |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-3.2.2/lib                |
| OpenSSL Version             | OpenSSL 1.1.1w  11 Sep 2023                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 15.3                                        |
| Swift Version               | 5.10                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

import './helper.rb'
import './global_lanes.rb'
import './private_lanes.rb'

default_platform(:android)

platform :android do
  desc 'Build the Android APP for testing and upload it to firebase'
  lane :testing do |options|
    release = patch_versions(version: options[:version], build_type: 'Debug')

    update_version_file(release)
    commit_files(release) if UI.confirm('Do you want to commit the version files?')
    build_ionic(options)
    gradle(
      task: 'clean',
      project_dir: './android/',
      print_command_output: false
    )
    gradle(
      task: 'assemble',
      build_type: 'Debug',
      project_dir: './android/',
      print_command_output: false
    )

    testing_upload(options) if options[:upload]
  end

  desc 'Builds the Android APP for release'
  lane :release do |options|
    release = patch_versions(version: options[:version], build_type: 'Release', hotfix: options[:hotfix])

    update_version_file(release)
    commit_files(release) if UI.confirm('Do you want to commit the version files?')
    build_ionic(env: 'production')
    gradle(
      task: 'clean',
      project_dir: './android/',
      print_command_output: false
    )
    gradle(
      task: 'bundle',
      build_type: 'Release',
      project_dir: './android/',
      print_command_output: false
    )

    if UI.confirm("Are you sure to upload the version #{release[:version]} (#{release[:code]}) to the store?")
      upload_to_play_store(
        release_status: 'draft',
        skip_upload_apk: true,
        validate_only: true,
        skip_upload_metadata: true,
        skip_upload_images: true,
        skip_upload_screenshots: true
      )
    end
  end

  private_lane :patch_versions do |options|
    release = get_version(options[:version], :android, options[:build_type], hotfix: options[:hotfix])

    android_set_version_name(version_name: release[:version], gradle_file: './android/app/build.gradle')
    android_set_version_code(version_code: release[:code], gradle_file: './android/app/build.gradle')
    UI.success "Android: Patched version #{release[:version]} and code #{release[:code]}"
    release
  end
end

platform :ios do
  desc 'Build the iOS APP and upload it to firebase'
  lane :testing do |options|
    release = patch_versions(version: options[:version], build_type: 'Debug')

    update_version_file(release)
    commit_files(release) if UI.confirm('Do you want to commit the version files?')
    build_ionic(options)
    build_ios_app(
      workspace: './ios/App/App.xcworkspace',
      scheme: 'DirectAPM',
      silent: true,
      export_method: 'development',
      output_directory: './ios/App/'
    )
    testing_upload(options) if options[:upload]
  end

  desc 'Builds the iOS APP for release'
  lane :release do |options|
    release = patch_versions(version: options[:version], build_type: 'Release', hotfix: options[:hotfix])

    update_version_file(release)
    commit_files(release) if UI.confirm('Do you want to commit the version files?')
    build_ionic(env: 'production')
    enable_automatic_code_signing(path: './ios/App/App.xcodeproj')
    build_ios_app(
      workspace: './ios/App/App.xcworkspace',
      scheme: 'DirectAPM',
      silent: true,
      export_method: 'app-store',
      output_directory: './ios/App/'
    )
    if UI.confirm('Are you sure to upload the build to the store?')
      upload_to_app_store(
        skip_metadata: true,
        skip_screenshots: true,
        automatic_release: false,
        include_in_app_purchases: false
      )
    end
  end

  private_lane :patch_versions do |options|
    release = get_version(options[:version], :ios, options[:build_type], hotfix: options[:hotfix])

    increment_version_number(version_number: release[:version], xcodeproj: './ios/App/App.xcodeproj')
    increment_build_number(build_number: release[:code], xcodeproj: './ios/App/App.xcodeproj')
    UI.success "iOS: Patched version #{release[:version]} and code #{release[:code]}"
    release
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file('./android/play-store-key.json') # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
package_name('....') # e.g. com.krausefx.app

app_identifier '....' # The bundle identifier of your app
apple_id '...' # Your Apple Developer Portal username
itc_team_name '...'

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.220.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                    | Version | Update-Status |
| ----------------------------------------- | ------- | ------------- |
| fastlane-plugin-firebase_app_distribution | 0.9.0   | ✅ Up-To-Date  |
| fastlane-plugin-versioning_android        | 0.1.1   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                         | Version      |
| ------------------------------------------- | ------------ |
| error_highlight                             | 0.5.1        |
| did_you_mean                                | 1.6.3        |
| syntax_suggest                              | 1.0.2        |
| executable-hooks                            | 1.7.1        |
| bundler-unload                              | 1.0.2        |
| rubygems-bundler                            | 1.4.5        |
| bundler                                     | 2.5.6        |
| pathname                                    | 0.2.1        |
| rake                                        | 13.2.1       |
| base64                                      | 0.2.0        |
| nkf                                         | 0.2.0        |
| rexml                                       | 3.2.6        |
| CFPropertyList                              | 3.0.7        |
| bigdecimal                                  | 3.1.3        |
| concurrent-ruby                             | 1.2.3        |
| connection_pool                             | 2.4.1        |
| ruby2_keywords                              | 0.0.5        |
| drb                                         | 2.1.1        |
| i18n                                        | 1.14.4       |
| minitest                                    | 5.16.3       |
| mutex_m                                     | 0.1.2        |
| tzinfo                                      | 2.0.6        |
| activesupport                               | 7.1.3.2      |
| public_suffix                               | 4.0.7        |
| addressable                                 | 2.8.6        |
| httpclient                                  | 2.8.3        |
| json                                        | 2.7.2        |
| algoliasearch                               | 1.27.5       |
| artifactory                                 | 3.0.17       |
| ast                                         | 2.4.2        |
| atomos                                      | 0.1.3        |
| aws-eventstream                             | 1.3.0        |
| aws-partitions                              | 1.909.0      |
| aws-sigv4                                   | 1.8.0        |
| jmespath                                    | 1.6.2        |
| aws-sdk-core                                | 3.191.6      |
| aws-sdk-kms                                 | 1.78.0       |
| aws-sdk-s3                                  | 1.146.1      |
| babosa                                      | 1.0.4        |
| backport                                    | 1.2.0        |
| benchmark                                   | 0.3.0        |
| claide                                      | 1.1.0        |
| fuzzy_match                                 | 2.0.4        |
| nap                                         | 1.1.0        |
| netrc                                       | 0.11.0       |
| ffi                                         | 1.16.3       |
| ethon                                       | 0.16.0       |
| typhoeus                                    | 1.4.1        |
| cocoapods-core                              | 1.15.2       |
| cocoapods-deintegrate                       | 1.0.5        |
| cocoapods-downloader                        | 2.1          |
| cocoapods-plugins                           | 1.0.0        |
| cocoapods-search                            | 1.0.1        |
| cocoapods-trunk                             | 1.6.0        |
| cocoapods-try                               | 1.2.0        |
| colored2                                    | 3.1.2        |
| escape                                      | 0.0.4        |
| fourflusher                                 | 2.3.1        |
| gh_inspector                                | 1.1.3        |
| molinillo                                   | 0.8.0        |
| ruby-macho                                  | 2.5.1        |
| nanaimo                                     | 0.3.0        |
| xcodeproj                                   | 1.24.0       |
| cocoapods                                   | 1.15.2       |
| colored                                     | 1.2          |
| highline                                    | 2.0.3        |
| commander                                   | 4.6.0        |
| stringio                                    | 3.1.0        |
| psych                                       | 5.1.2        |
| rdoc                                        | 6.6.2        |
| io-console                                  | 0.7.2        |
| reline                                      | 0.4.3        |
| irb                                         | 1.12.0       |
| debug                                       | 1.9.1        |
| declarative                                 | 0.0.20       |
| diff-lcs                                    | 1.5.1        |
| digest-crc                                  | 0.6.5        |
| domain_name                                 | 0.6.20240107 |
| dotenv                                      | 2.8.1        |
| e2mmap                                      | 0.1.0        |
| emoji_regex                                 | 3.2.3        |
| excon                                       | 0.110.0      |
| faraday-em_http                             | 1.0.0        |
| faraday-em_synchrony                        | 1.0.0        |
| faraday-excon                               | 1.1.0        |
| faraday-httpclient                          | 1.0.1        |
| multipart-post                              | 2.4.0        |
| faraday-multipart                           | 1.0.4        |
| faraday-net_http                            | 1.0.1        |
| faraday-net_http_persistent                 | 1.2.0        |
| faraday-patron                              | 1.0.0        |
| faraday-rack                                | 1.0.0        |
| faraday-retry                               | 1.0.3        |
| faraday                                     | 1.10.3       |
| http-cookie                                 | 1.0.5        |
| faraday-cookie_jar                          | 0.0.7        |
| faraday_middleware                          | 1.2.0        |
| fastimage                                   | 2.3.1        |
| jwt                                         | 2.8.1        |
| multi_json                                  | 1.15.0       |
| os                                          | 1.1.4        |
| signet                                      | 0.19.0       |
| googleauth                                  | 1.8.1        |
| mini_mime                                   | 1.1.5        |
| trailblazer-option                          | 0.1.2        |
| uber                                        | 0.1.0        |
| representable                               | 3.2.0        |
| retriable                                   | 3.1.2        |
| google-apis-core                            | 0.11.3       |
| google-apis-androidpublisher_v3             | 0.54.0       |
| google-apis-playcustomapp_v1                | 0.13.0       |
| google-cloud-env                            | 1.6.0        |
| google-apis-iamcredentials_v1               | 0.17.0       |
| google-apis-storage_v1                      | 0.31.0       |
| google-cloud-errors                         | 1.4.0        |
| google-cloud-core                           | 1.7.0        |
| google-cloud-storage                        | 1.47.0       |
| mini_magick                                 | 4.12.0       |
| naturally                                   | 2.2.1        |
| optparse                                    | 0.4.0        |
| plist                                       | 3.7.1        |
| rubyzip                                     | 2.3.2        |
| security                                    | 0.1.5        |
| simctl                                      | 1.6.10       |
| terminal-notifier                           | 2.0.0        |
| unicode-display_width                       | 2.5.0        |
| terminal-table                              | 3.0.2        |
| tty-screen                                  | 0.8.2        |
| tty-cursor                                  | 0.7.1        |
| tty-spinner                                 | 0.9.3        |
| word_wrap                                   | 1.0.0        |
| rouge                                       | 2.0.7        |
| xcpretty                                    | 0.3.0        |
| xcpretty-travis-formatter                   | 1.0.1        |
| google-apis-firebaseappdistribution_v1      | 0.3.0        |
| google-apis-firebaseappdistribution_v1alpha | 0.2.0        |
| fastlane-plugin-firebase_app_distribution   | 0.9.0        |
| fastlane-plugin-versioning_android          | 0.1.1        |
| jaro_winkler                                | 1.5.6        |
| kramdown                                    | 2.4.0        |
| kramdown-parser-gfm                         | 1.1.0        |
| language_server-protocol                    | 3.17.0.3     |
| racc                                        | 1.7.3        |
| nokogiri                                    | 1.16.3       |
| parallel                                    | 1.24.0       |
| parser                                      | 3.3.0.5      |
| prettier_print                              | 1.2.1        |
| prism                                       | 0.24.0       |
| rainbow                                     | 3.1.1        |
| rbs                                         | 2.8.4        |
| regexp_parser                               | 2.9.0        |
| reverse_markdown                            | 2.1.1        |
| rubocop-ast                                 | 1.31.2       |
| ruby-progressbar                            | 1.13.0       |
| rubocop                                     | 1.62.1       |
| rubocop-capybara                            | 2.20.0       |
| rubocop-factory_bot                         | 2.25.1       |
| rubocop-rspec                               | 2.27.1       |
| rubocop-discourse                           | 3.7.1        |
| sorbet-runtime                              | 0.5.11295    |
| ruby-lsp                                    | 0.14.6       |
| thor                                        | 1.3.1        |
| tilt                                        | 2.3.0        |
| yard                                        | 0.9.36       |
| solargraph                                  | 0.50.0       |
| syntax_tree                                 | 6.2.0        |
| syntax_tree-disable_ternary                 | 1.0.0        |
</details>

</details>

My goal is just to have a draft version on the play store that I can complete and release, does someone have an idea why the AAb is not uploaded on the Play store ?

Thanks in advance 👍

**Comment by mataku** [comment, marked as answer]
You called update_to_play_store with `validate_only: true`, which means `Validates the current edit - does not change data on Google Play`.

If false, the request would be made to the Google Play API. The default value is false:
https://docs.fastlane.tools/actions/upload_to_play_store/

**Reply by alexandrebedel** [reply]
Well 😁 it make sense now, thank you so much
